docs/ADD-APP.md

```diff
@@ -0,0 +1,149 @@
+# Add Django app
+This project uses Django apps to organise code. This includes both front- and backend code. An app is created for each type of widget. This paragraph describes how one creates such an app and configure it.
+
+>[!IMPORTANT]
+> Before following these steps, ensure that your Django-server is installed and configured. You can find the instructions for this in the (development setup docs)[https://github.com/MirrorBoys/TheMirror/blob/main/docs/DEVELOPMENT.md#development-setup].
+>[!IMPORTANT]
+> Ensure that you are following these steps in your Python environment.
+
+## Name the app
+Try to use a generic name such as `mailWidget` instead of `gmailWidget`. This makes it easier to switch API providers without having to rename a lot of files and references. Save this app in a variable so we can use it in the next steps. `appType` will be used to name the app's template file.
+```shell
+appName="<name of your app>"
+appType="<type of your app>"
+```
+>[!NOTE]
+> This variable is only created for your current shell. It will be deleted if you exit your shell.
+
+## Create app
+In these steps we will use Django's `manage.py` applicaton to create the app and it's folders.
+
+1. Create the app
+```shell
+python3 ~/TheMirror/src/smartMirrorProject/manage.py startapp $appName
+```
+2. Add the app to the Django project settings file.
+```shell
+sudo nano ~/TheMirror/src/smartMirrorProject/smartMirrorProject/settings.py
+```
+
+```shell
+...
+INSTALLED_APPS = [
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'homePage',
+    'musicWidget',
+    'newsWidget',
+    'travelWidget',
+    'weatherWidget',
+
+    # Add the following line
+    '<name of your app>',
+]
+...
+```
+
+## Configure app
+After creating the app we need to configure it.
+
+### Adding static files
+We create the folders and files which will store the static files.
```

Comment:
Vraagje. Als je een app maakt, is het dan niet zo dat je ook zelf de static folder aan moet maken en de templates folder? Dan moest ik wel.